src/api/BaseServiceApi.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpMethod, IBaseServiceApi } from '@/types/serviceTypes'
+import { ApiError } from './BaseError'
+
+export class BaseServiceApi implements IBaseServiceApi {
+  baseUrl: string
+  error: typeof ApiError
+  constructor(baseUrl: string, error = ApiError) {
+    this.baseUrl = baseUrl
+    this.error = error
+  }
+
+  private async handleFetch<Type>(method: HttpMethod, url: string, body?: Type | Type[]) {
+    try {
```

Comment:
const response = await fetch(this.baseUrl + url, {
        method: method,
        headers: {
          'Content-Type': 'application/json',
        },
        body: body ? JSON.stringify(body) : undefined,
      })


if (!response.ok) {
        throw new this.error('error on retrieving data', response.status, response.statusText)
      }

try {
 return  response.json()
} catch (e) {
throw new this.error('Failed when parsing JSON object', 400, 'description')
}